File: VB/VB/WinSolution.Module/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _propertyName1 As String
		<RuleRequiredField("1", DefaultContexts.Save)> _
		Public Property PropertyName1() As String
			Get
				Return _propertyName1
			End Get
			Set(ByVal value As String)
				SetPropertyValue("PropertyName1", _propertyName1, value)
			End Set
		End Property
		Private _propertyName2 As String
		<RuleRequiredField("2", DefaultContexts.Save)> _
		Public Property PropertyName2() As String
			Get
				Return _propertyName2
			End Get
			Set(ByVal value As String)
				SetPropertyValue("PropertyName2", _propertyName2, value)
			End Set
		End Property
	End Class

End Namespace

File: CS/2/WinSolution.Module/DomainObject1.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinSolution.Module {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session) : base(session) { }
        private string _propertyName1;
        [RuleRequiredField("1", DefaultContexts.Save)]
        public string PropertyName1 {
            get { return _propertyName1; }
            set { SetPropertyValue("PropertyName1", ref _propertyName1, value); }
        }
        private string _propertyName2;
        [RuleRequiredField("2", DefaultContexts.Save)]
        public string PropertyName2 {
            get { return _propertyName2; }
            set { SetPropertyValue("PropertyName2", ref _propertyName2, value); }
        }
    }

}
